pkg/pool-linear/test/AaveLinearPoolFactory.test.ts

```diff
@@ -72,6 +78,14 @@ describe('AaveLinearPoolFactory', function () {
       expect(await pool.getVault()).to.equal(vault.address);
     });
 
+    it('checks the factory version', async () => {
+      expect(await factory.version()).to.equal(version);
+    });
+
+    it('checks the pool version', async () => {
+      expect(await pool.version()).to.equal(version);
+    });
```

Comment:
Don't you think it's weird that the factory returns the name of the pool?

Reply:
If we're just storing values in storage, we could drop the whole version provider idea, and have the factory pass a version to the pool. The only issue is we'd either have to a) generate the json on the fly (and solidity is bad) or b) pass both the factory and pool version to the factory constructor.

Reply:
> Don't you think it's weird that the factory returns the name of the pool?

Hmm not really. When we do a deployment, we think of e.g. `ManagedPool`, `ComposableStablePool`, etc; the fact that the pools actually come from factories is sort of implicit.

Generating the JSON in solidity doesn't sound very good. 

I'd say that if pools of a given version will always come from the same factory, I don't see a reason to make a difference between factory / pool version. We could rename the method in the factory if it sounds misleading.

Otherwise, I guess we can give the factory the two names.